third_party/CMakeLists.txt

```diff
@@ -65,14 +65,25 @@ if (${AMBER_USE_LOCAL_VULKAN})
   add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vulkan-validationlayers)
 endif()
 
+if (${AMBER_ENABLE_VK_DEBUGGING})
+  set(CPPDAP_JSON_DIR ${CMAKE_CURRENT_SOURCE_DIR}/json)
+  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/cppdap)
+endif()
+
 if (${AMBER_ENABLE_SWIFTSHADER})
-  set(SWIFTSHADER_BUILD_EGL FALSE CACHE BOOL FALSE)
-  set(SWIFTSHADER_BUILD_GLESv2 FALSE CACHE BOOL FALSE)
-  set(SWIFTSHADER_BUILD_GLES_CM FALSE CACHE BOOL FALSE)
-  set(SWIFTSHADER_BUILD_VULKAN TRUE CACHE BOOL TRUE)
-  set(SWIFTSHADER_BUILD_SAMPLES FALSE CACHE BOOL FALSE)
-  set(SWIFTSHADER_BUILD_TESTS FALSE CACHE BOOL FALSE)
-  set(SWIFTSHADER_WARNINGS_AS_ERRORS FALSE CACHE BOOL FALSE)
+  set(SWIFTSHADER_BUILD_EGL FALSE)
```

Comment:
Why remove the CACHE BOOL. Those seem to be needed to keep cmake happy. If you don't cache it then option overrides it, or something wonky.

Reply:
I changed SwiftShader so that it'll check for these to be set before promoting them up to options. This way they won't pollute the amber options, when they must not be changed.